comparators/dragged_highway_waterway.js

```diff
@@ -3,29 +3,39 @@
 module.exports = draggedHighwayWaterway;
 var turfPoint = require('turf-point');
 var turfDistance = require('turf-distance');
-var threshold = 10;
+
+function hasDrag(feature) {
+
+  // Not interesting if there isn't a feature.
+  if (!feature) return false;
+
+  // Not interesting if there are not properties and geometries.
+  var properties = feature.properties;
+  var geometry = feature.geometry;
+  if (!properties || !geometry) return false;
+
+  // Not interesting if not a highway or a waterway.
+  if (!(properties.hasOwnProperty('highway') || properties.hasOwnProperty('waterway'))) return false;
+
+  // Not interesting if has required geometry.
+  if ((properties['osm:type'] !== 'way') || (geometry.coordinates.length <= 1)) return false;
+
+  var threshold = 10;  // In kilometers.
+  for (var i = 0; i < geometry.coordinates.length - 1; i++) {
+    var point1 = turfPoint(geometry.coordinates[i]);
+    var point2 = turfPoint(geometry.coordinates[i + 1]);
+
+    var distance = turfDistance(point1, point2, 'kilometers');
+    // The feature has a drag if distance is greater than threshold.
+    if (distance > threshold) return true;
+  }
+}
 function draggedHighwayWaterway(newVersion, oldVersion) {
   var result = {};
 
-  if (!newVersion) {
-    return result;
-  }
+  var newVersionDragged = hasDrag(newVersion);
+  var oldVersionDragged = hasDrag(oldVersion);
 
-  if (newVersion.properties &&
-    (newVersion.properties.hasOwnProperty('highway') ||
-    newVersion.properties.hasOwnProperty('waterway')) &&
-    newVersion.properties['osm:type'] === 'way' &&
-    newVersion.geometry &&
-    newVersion.geometry.hasOwnProperty('coordinates') &&
-    newVersion.geometry.coordinates.length > 1) {
-    for (var i = 0; i < newVersion.geometry.coordinates.length - 1; i++) {
-      var point1 = turfPoint(newVersion.geometry.coordinates[i]);
-      var point2 = turfPoint(newVersion.geometry.coordinates[i + 1]);
-      var distance = turfDistance(point1, point2, 'kilometers');
-      if (distance > threshold) {
-        result['result:dragged_highway_waterway'] = true;
-      }
-    }
-  }
+  if (newVersionDragged && !oldVersionDragged) result = {'result:dragged_highway_waterway': true};
```

Comment:
This is amazing @amishas157, thank you. 😃 

Looks like the previous version of the comparator did this right except for one thing, not to do anything if the geometry of the newVersion and the oldVersion are the same. Will revert the compare function to it's earlier state and add that check on top.